### PR TITLE
feat(device_socket): init

### PR DESCRIPTION
**Pull request by noaccOS**
implement a socket for the device to connect to, using websocket

depends on ~#6~  #11 and includes its commits

### CODE REVIEW

File: lib/edgehog_device_forwarder_web/device_socket.ex [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  alias EdgehogDeviceForwarder.Tokens
+  alias EdgehogDeviceForwarder.WebSockets
+  require Logger
+
+  @type state :: Phoenix.Socket.Transport.state()
+
+  @spec child_spec(keyword) :: :ignore
+  def child_spec(_opts) do
+    :ignore
+  end
```
Comment (rbino) [inline lib/edgehog_device_forwarder_web/device_socket.ex:22]: Since it's the first time I see this used, can you detail why is this needed?

**Review by rbino — CHANGES_REQUESTED** [review]

File: lib/edgehog_device_forwarder_web/device_socket.ex (hunk 1 above)
Comment (noaccOS) [reply lib/edgehog_device_forwarder_web/device_socket.ex:22]: `child_spec/1` is required from the `Phoenix.Socket.Transport` specification.
From the [documentation](https://hexdocs.pm/phoenix/Phoenix.Socket.Transport.html#c:child_spec/1), 

> `:ignore` means no child spec is necessary for this socket

Which is the case for us, as we don't spawn any additional process from the socket

**Review by rbino — APPROVED** [review]
